### PR TITLE
docs: update migration guide for nuxt 3

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/framework/issues/3769

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)

### 📚 Description

This is a WIP PR for the Nuxt 3 migration guide.

Preview: https://deploy-preview-3819--nuxt3-docs.netlify.app/docs/migration/overview

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
Shall we move the migration section to "Docs"? Getting started could be preserved for Nuxt 3 starters and simple this way.

**Comment by danielroe** [comment]
This is very much a first stab at the migration guide. Would value any input at this point, though of course we can also merge + iterate as @pi0 suggested.

File: docs/content/3.docs/1.migration/20.module-authors.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Modules
+
+## Module Compatibility
+
+Nuxt 2 modules should be forward compatible with Nuxt 3 as long as users migrate to Bridge, or if modules are already following guidelines.
+
+All (upcoming) modules made with `@nuxt/kit` should be backward compatible with Nuxt 2 projects (even without Bridge) as long as they are not depending on a Nuxt 3/Bridge-only feature.
+
```
Comment (antfu) [inline docs/content/3.docs/1.migration/20.module-authors.md:8]:
```suggestion

You could check [Nuxt Modules](https://modules.nuxtjs.org/) for modules compatibility and as a reference for the project setup from other modules.
```

**Review by pi0 — APPROVED** [review]
Added small updates. Rests lgtm. Fantastic work!

**Pull request merged**